cmake/CodeCoverage.cmake

```diff
@@ -0,0 +1,708 @@
+# Copyright (c) 2012 - 2017, Lars Bilke
```

Comment:
this is a nice helper cmake module that makes adding coverage to our project easy. it looks to be 3 clause bsd. i dont think that is a problem but if you'd rather not add this, i can do the coverage configuration and cmake targets manually. just let me know

Reply:
I'm not a license expert but it seems that the only different between MIT and this one is 
```
# 3. Neither the name of the copyright holder nor the names of its contributors
#    may be used to endorse or promote products derived from this software without
#    specific prior written permission.
```
so it should be ok.